src/main/java/reactor/kafka/sender/KafkaSender.java

```diff
@@ -44,8 +44,19 @@
      *        after the sender is created will not be used by the sender.
      * @return new instance of Kafka sender
      */
-    public static <K, V> KafkaSender<K, V> create(SenderOptions<K, V> options) {
-        return new DefaultKafkaSender<>(ProducerFactory.INSTANCE, options);
+    static <K, V> KafkaSender<K, V> create(SenderOptions<K, V> options) {
+        return create(DefaultProducerFactory.INSTANCE, options);
+    }
+
+    /**
+     * Creates a Kafka sender that appends records to Kafka topic partitions.
+     * @param factory Custom KafkaProducer factory.
+     * @param options Configuration options of this sender. Changes made to the options
+     *        after the sender is created will not be used by the sender.
+     * @return new instance of Kafka sender
+     */
+    static <K, V> KafkaSender<K, V> create(ProducerFactory factory, SenderOptions<K, V> options) {
```

Comment:
Let's just keep that method without an introduction of new interfaces

Reply:
done